bluefog/torch/optimizers.py

```diff
@@ -335,42 +352,42 @@ def __init__(self, params, model, communication_type, num_steps_per_communicatio
 
     def _register_hooks(self):
         for model in self._models:
-            for parent_name, layer in _named_leaf_module(model):
-                layer.register_forward_hook(self._make_hook(parent_name))
+            model.register_forward_hook(self._make_hook())
```

Comment:
Add a comment here about why model forward is sufficient

Reply:
Done.